src/ImageSharp/Formats/Tga/TgaEncoderCore.cs

```diff
@@ -0,0 +1,356 @@
+        /// <param name="stream">The <see cref="Stream"/> to write to.</param>
+        /// <param name="pixels">The <see cref="Buffer2D{TPixel}"/> containing pixel data.</param>
+        private void Write24Bit<TPixel>(Stream stream, Buffer2D<TPixel> pixels)
+            where TPixel : struct, IPixel<TPixel>
+        {
+            using (IManagedByteBuffer row = this.AllocateRow(pixels.Width, 3))
+            {
+                for (int y = pixels.Height - 1; y >= 0; y--)
+                {
+                    Span<TPixel> pixelSpan = pixels.GetRowSpan(y);
+                    PixelOperations<TPixel>.Instance.ToBgr24Bytes(
+                        this.configuration,
+                        pixelSpan,
+                        row.GetSpan(),
+                        pixelSpan.Length);
+                    stream.Write(row.Array, 0, row.Length());
+                }
+            }
+        }
+
+        /// <summary>
+        /// Writes the 32bit pixels uncompressed to the stream.
+        /// </summary>
+        /// <typeparam name="TPixel">The pixel format.</typeparam>
+        /// <param name="stream">The <see cref="Stream"/> to write to.</param>
+        /// <param name="pixels">The <see cref="Buffer2D{TPixel}"/> containing pixel data.</param>
+        private void Write32Bit<TPixel>(Stream stream, Buffer2D<TPixel> pixels)
+            where TPixel : struct, IPixel<TPixel>
+        {
+            using (IManagedByteBuffer row = this.AllocateRow(pixels.Width, 4))
+            {
+                for (int y = pixels.Height - 1; y >= 0; y--)
+                {
+                    Span<TPixel> pixelSpan = pixels.GetRowSpan(y);
+                    PixelOperations<TPixel>.Instance.ToBgra32Bytes(
+                        this.configuration,
+                        pixelSpan,
+                        row.GetSpan(),
+                        pixelSpan.Length);
+                    stream.Write(row.Array, 0, row.Length());
+                }
+            }
+        }
+
+        /// <summary>
+        /// Convert the pixel values to grayscale using ITU-R Recommendation BT.709.
+        /// </summary>
+        /// <param name="sourcePixel">The pixel to get the luminance from.</param>
+        [MethodImpl(InliningOptions.ShortMethod)]
+        public static int GetLuminance<TPixel>(TPixel sourcePixel)
+            where TPixel : struct, IPixel<TPixel>
+        {
+            var vector = sourcePixel.ToVector4();
+            return GetLuminance(ref vector);
+        }
+
+        /// <summary>
+        /// Convert the pixel values to grayscale using ITU-R Recommendation BT.709.
+        /// </summary>
+        /// <param name="vector">The vector to get the luminance from.</param>
+        [MethodImpl(InliningOptions.ShortMethod)]
+        public static int GetLuminance(ref Vector4 vector)
+            => (int)MathF.Round(((.2126F * vector.X) + (.7152F * vector.Y) + (.0722F * vector.Y)) * (256 - 1));
```

Comment:
There's a bug here. Should be `.722F * vector.Z`

I wonder? Could this be calculated as the result of the dot product of the two vectors?

If not, probably best to use 

`(int)MathF.Round(ImageMaths.Get16BitBT709Luminance(vector.X, vector.Y, vector.Z) * (256 - 1));`

Reply:
argh, good spot  :+1:
how could i missed that? 

Dot product is the same i could use that. Unfortunately the same issue is in the histogram equalization!

Reply:
i have moved the luminance calculation with the dotproduct to ImageMaths. Both the TgaEncoder and the histogram equalization now use this.
As expected the histogram tests fail now. I will provide new reference images tomorrow.

Reply:
i have updated the testimages in [PR 13](https://github.com/SixLabors/Imagesharp.Tests.Images/pull/13)